tools/syz-reporter/reporter.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2020 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+// syz-reporter creates table information from crashes.
+// Useful tool together with tools/syz-crush to collect
+// results from the reproducer runs.
+
+// Nice extension to this would be to accept multiple configurations and
+// then collect table from all the different workdirectories. This would allow easy comparison
+// if different kernel version have same BUGs.
+
+package main
+
+import (
+	"flag"
+	"io"
+	"io/ioutil"
+	"log"
+
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/google/syzkaller/pkg/html"
+	"github.com/google/syzkaller/pkg/mgrconfig"
+	"github.com/google/syzkaller/pkg/osutil"
+)
+
+var (
+	flagConfig = flag.String("config", "", "configuration file")
+)
+
+type UISummaryData struct {
+	Name    string
+	Crashes []*UICrashType
+	Workdir string
+}
+
+type UICrashType struct {
+	Description string
+	ID          string
+	Count       int
+	Tags        map[string]string
+	Crashes     []*UICrash
+}
+
+type UICrash struct {
+	Index  int
+	Log    string
+	Report string
+}
+
+func main() {
+	flag.Parse()
+	cfg, err := mgrconfig.LoadFile(*flagConfig)
+	if err != nil {
+		log.Fatalf("%v", err)
+	}
+	httpSummary(os.Stdout, cfg)
+}
+
+func httpSummary(w io.Writer, cfg *mgrconfig.Config) {
+	data := &UISummaryData{
+		Name:    cfg.Name,
+		Workdir: cfg.Workdir,
+	}
+
+	var err error
+	if data.Crashes, err = collectCrashes(cfg.Workdir); err != nil {
+		log.Printf("failed to collect crashes: %v", err)
+		return
```

Comment:
This will make it exit with 0 status. I think it makes sense to return an error from this function as:
```
return fmt.Errrof("failed to collect crashes: %v", err)
```
(here and below)
and then do log.Fatalf in the caller.
WIll make it fail and also consistent with error handling for mgrconfig.LoadFile.